### PR TITLE
Only run dmsetup mknodes if dmsetup install is not skipped

### LINKED ISSUE #126: Run dmsetup for Ubuntu 16.04 is run on all Ubuntu versions
Currently the command with the name: Run dmsetup for Ubuntu 16.04 is run for all versions. This is happening because the when is checking against None where it should be checking against the skipped key in the result dictionary.

### PR DESCRIPTION
**Pull request by rjduffner**
This fixes #126

@angstwad

### CODE REVIEW

**Comment by rjduffner** [comment]
This change results in this for runs against 12.04

```
TASK [docker.ubuntu : Install dmsetup for Ubuntu 16.04] ************************
task path: /Users/robertduffner/SurveyMonkey/docker.ubuntu/tasks/main.yml:18
skipping: [ubuntu-1204] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

TASK [docker.ubuntu : Read uname] **********************************************
task path: /Users/robertduffner/SurveyMonkey/docker.ubuntu/tasks/main.yml:27
changed: [ubuntu-1204] => {"changed": true, "cmd": "uname -r", "delta": "0:00:00.002082", "end": "2017-01-03 21:23:41.408311", "rc": 0, "start": "2017-01-03 21:23:41.406229", "stderr": "", "stdout": "3.2.0-118-virtual", "stdout_lines": ["3.2.0-118-virtual"], "warnings": []}

TASK [docker.ubuntu : Install linux-image-extra-* packages to enable AuFS driver] ***
task path: /Users/robertduffner/SurveyMonkey/docker.ubuntu/tasks/main.yml:31
skipping: [ubuntu-1204] => (item=[])  => {"changed": false, "item": [], "skip_reason": "Conditional check failed", "skipped": true}

TASK [docker.ubuntu : debug] ***************************************************
task path: /Users/robertduffner/SurveyMonkey/docker.ubuntu/tasks/main.yml:38
ok: [ubuntu-1204] => {
    "msg": "dmsetup_result: {'skip_reason': u'Conditional check failed', 'skipped': True, 'changed': False} Success: True"
}

TASK [docker.ubuntu : Run dmsetup for Ubuntu 16.04] ****************************
task path: /Users/robertduffner/SurveyMonkey/docker.ubuntu/tasks/main.yml:41
skipping: [ubuntu-1204] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}
```

Note: I added a debug line in there to see the structure of dmsetup_result. It has been removed in the final PR.

**Comment by rjduffner** [comment]
I also added a second commit to run an apt-get update when installing the AuFS driver. This was failing to install when I was testing because the default apt-get cache was out of date on the base box.

File: tasks/main.yml [hunk 1]
```diff
@@ -29,15 +29,18 @@
...
+    update_cache: yes
   with_items:
     - linux-image-extra-{{ uname_output.stdout }}
     - linux-image-extra-virtual
   when: ansible_distribution_version|version_compare(14.04, '>=')
 
 - name: Run dmsetup for Ubuntu 16.04
   command: dmsetup mknodes
-  when: dmsetup_result
+  when: dmsetup_result.get("skipped", False) != True
```
Comment (angstwad) [inline tasks/main.yml]: I think this can just be `not dmsetup_result.skipped | default(True)`

File: tasks/main.yml [hunk 2]
```diff
@@ -29,15 +29,18 @@
   register: uname_output
 
 - name: Install linux-image-extra-* packages to enable AuFS driver
-  apt: pkg={{ item }} state=present
+  apt:
+    pkg: "{{ item }}"
+    state: present
+    update_cache: yes
```
Comment (angstwad) [inline tasks/main.yml]: Can you use cache valid time like here: https://github.com/rjduffner/docker.ubuntu/blob/bd874bd6c43090a062943e6b227c543747cac892/tasks/main.yml#L151

File: tasks/main.yml (hunk 1 above)
Comment (angstwad) [inline tasks/main.yml]: ...but double check the logic. Can't remember if the default needs to be True or False, if the not negates the entire expression after eval'ed by default filter

**Comment by angstwad** [comment]
@rjduffner Please take a look at the comments, rebase, and test please

File: tasks/main.yml (hunk 1 above)
Comment (angstwad) [inline tasks/main.yml]: Wait, why not just use `changed?`

Comment (angstwad) [inline tasks/main.yml]: When you rebase you'll see that this is already addressed so I think you can just remove this.

**Comment by screwlewse** [comment]
@angstwad Thanks for the feedback.   RJ is off on vacation but I'll have him check it as soon as he's back.  Thanks for checking this over!

**Comment by rjduffner** [comment]
@angstwad it should be all set to go. Let me know if you see anything else.

**Comment by angstwad** [comment]
Thanks!

**Pull request merged**